frontend/models/product.ts

```diff
@@ -0,0 +1,159 @@
+import { IFIXIT_ORIGIN } from '@config/env';
+import { filterNullableItems, invariant } from '@helpers/application-helpers';
+import { isRecord } from '@ifixit/helpers';
+import {
+   ShopCredentials,
+   getShopifyStorefrontSdk,
+   MoneyV2,
+   CurrencyCode,
+} from '@lib/shopify-storefront-sdk';
+
+export async function findProduct(shop: ShopCredentials, handle: string) {
+   const storefront = getShopifyStorefrontSdk(shop);
+
+   const response = await storefront.findProduct({
+      handle,
+   });
+   if (response.product == null) {
+      return null;
+   }
+   const variants = response.product.variants.nodes.map((variant) => {
+      return {
+         ...variant,
+         formattedPrice: formatPrice(variant.priceV2),
+         formattedCompareAtPrice: variant.compareAtPriceV2
+            ? formatPrice(variant.compareAtPriceV2)
+            : null,
+         kitContents: variant.kitContents?.value ?? null,
+         note: variant.note?.value ?? null,
+         disclaimer: variant.disclaimer?.value ?? null,
+         warning: variant.warning?.value ?? null,
+         specifications: variant.specifications?.value ?? null,
+         warranty: variant.warranty?.value ?? null,
+      };
+   });
+   const variantSku = variants.find((variant) => variant.sku != null)?.sku;
+   if (variantSku == null) {
+      console.warn(`No sku found for product "${handle}"`);
+      return null;
+   }
+   const iFixitProductId = computeIFixitProductId(variantSku);
+   const reviewsData = await fetchProductReviews(
+      IFIXIT_ORIGIN,
+      iFixitProductId
+   );
+
+   return {
+      ...response.product,
+      iFixitProductId,
+      variants,
+      images: response.product.images.nodes.map((node) => node),
+      prop65WarningType: response.product.prop65WarningType?.value ?? null,
+      prop65Chemicals: response.product.prop65Chemicals?.value ?? null,
+      productVideos: response.product.productVideos?.value ?? null,
+      faqs: parseFaqs(response.product.faqs?.value),
+      reviewsData,
+   };
+}
+
+export type Product = NonNullable<Awaited<ReturnType<typeof findProduct>>>;
+
+export type ProductVariant = Product['variants'][0];
+
+function formatPrice(money: MoneyV2) {
+   switch (money.currencyCode) {
+      case CurrencyCode.Usd: {
+         return `$${money.amount}`;
+      }
+      default: {
+         return `${money.amount} ${money.currencyCode}`;
+      }
+   }
+}
+
+function parseFaqs(value: string | null | undefined) {
+   if (value == null) {
+      return [];
+   }
+   const faqs = JSON.parse(value);
+   if (!Array.isArray(faqs)) {
+      return [];
+   }
```

Comment:
First of all, right now this is a custom Shopify metafield that is not being populated by some iFixit api. So the only way to edit it is to go to the product page on Shopify e write the JSON there. 

<img width="780" alt="Screenshot 2022-08-24 at 11 45 00" src="https://user-images.githubusercontent.com/4640135/186387087-0201c1a8-1318-4149-8e3e-a25d58ad4099.png">

I'm not sure if we'll end up filling the metafield via other means, so I've not made too many assumptions here.

Since I receive plain JSON, I try to be defensive here and assert that I should receive an array. Potentially we might report errors to Sentry, but it would be a data entry issue, not a code issue.